File: OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignalBounceInterpolator.java
package com.onesignal;

class OneSignalBounceInterpolator implements android.view.animation.Interpolator {
    private double mAmplitude = 1;
    private double mFrequency = 10;

    OneSignalBounceInterpolator(double amplitude, double frequency) {
        mAmplitude = amplitude;
        mFrequency = frequency;
    }

    public float getInterpolation(float time) {
        return (float) (-1 * Math.pow(Math.E, -time / mAmplitude) * Math.cos(mFrequency * time) + 1);
    }
}

File: OneSignalSDK/onesignal/in-app-messages/src/main/java/com/onesignal/inAppMessages/internal/display/impl/OneSignalBounceInterpolator.kt
package com.onesignal.inAppMessages.internal.display.impl

import android.view.animation.Interpolator

internal class OneSignalBounceInterpolator(amplitude: Double, frequency: Double) : Interpolator {
    private var mAmplitude = 1.0
    private var mFrequency = 10.0

    override fun getInterpolation(time: Float): Float {
        return (
            -1 *
                Math.pow(
                    Math.E,
                    -time / mAmplitude,
                ) * Math.cos(mFrequency * time) + 1
            ).toFloat()
    }

    init {
        mAmplitude = amplitude
        mFrequency = frequency
    }
}
